test/test_manual.jl

```diff
@@ -59,6 +60,116 @@ isdir(outdir) && rm(outdir, recursive=true)
     end
   end
 
+  @testset "ParallelTreeMesh" begin
+    @testset "partition!" begin
+      @testset "mpi_nranks() = 2" begin
+        mock((Trixi.mpi_nranks) => () -> 2) do mpi_ranks
```

Comment:
If I understand correctly, the variable name `mpi_ranks` now refers to the mock function for calls to, e.g., `called_once_with`. However, it's also the name of one of the fields in `mesh.tree`, so that's kind of confusing. Maybe choose a different varname here, e.g., `mock_mpi_nranks`?